### PR TITLE
Make ts sdk better

### LINKED ISSUE #114: Fix Typescript sdk
A few fixes for the sdk before for a more stable alpha
* [ ] Combine middlewares at build time instead of execution time.
* [ ] Allow multiple request parameters.
* [ ] Fix ts-lint issues.

### PR DESCRIPTION
**Pull request by acbeni**
closes #114

### CODE REVIEW

File: languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ class BuilderTestCodeGenerator {
          * Specifies a path where the code generator should place generated code
          * */
         private val generatedCodePath = System.getenv("GENERATED_CODE_PATH")
-        private val apiPath : Path = Paths.get(if (userProvidedPath == null) "../api-spec/api.raml" else userProvidedPath)
+        private val apiPath : Path = Paths.get("/Users/abeniasaad/IdeaProjects/commercetools-api-reference/update-actions.raml")
```
Comment (katmatt) [inline languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt]: Please revert this change and use our `GENERATED_CODE_PATH` env var instead.

File: languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt [hunk 2]
```diff
@@ -64,7 +64,7 @@ class BuilderTestCodeGenerator {
     @Ignore
     @Test
     fun generateJavaCompleteModule() {
-        val generatorConfig = CodeGeneratorConfig(basePackageName = "com.commercetools", outputFolder = Paths.get(generatedCodePath))
+        val generatorConfig = CodeGeneratorConfig(basePackageName = "com.commercetools.api.generated", outputFolder = Paths.get("/Users/abeniasaad/IdeaProjects/commercetools-java-sdks/commercetools/commercetools-api-generated/src/main/java"))
```
Comment (katmatt) [inline languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt]: Same as above.

File: languages/typescript/src/test/resources/typescript-code/tslint-vscode.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (katmatt) [inline languages/typescript/src/test/resources/typescript-code/tslint-vscode.json]: Is this file automatically used by vscode? And why do we have to use a separate tslint file for vscode?

Comment (acbeni) [reply languages/typescript/src/test/resources/typescript-code/tslint-vscode.json]: Its actually just a copy paste from the node repo that i use to test the quality of my generated code, but as long as we don't have a correct ci, i just have to use it mannually wenever i do work on the node sdk 😢

File: languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt (hunk 1 above)
Comment (acbeni) [reply languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt]: fixed in 25da514

File: languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt (hunk 2 above)
Comment (acbeni) [reply languages/javalang/builder-renderer/java-builder-client/src/test/kotlin/io/vrap/codegen/languages/javalang/client/builder/BuilderTestCodeGenerator.kt]: fixed in 25da514
